### PR TITLE
CSL Preferences [WIP]

### PR DESCRIPTION
**Pull request by subhramit**
Halted due to architecture changes (#7 ).
### Mandatory checks

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/logic/openoffice/OpenOfficePreferences.java [hunk 1]
```diff
@@ -26,17 +26,22 @@ public class OpenOfficePreferences {
...
                                  List<String> externalStyles,
-                                 String currentStyle) {
+                                 String currentStyle, String currentStyleType,
+                                 String currentCslStyleName) {
         this.executablePath = new SimpleStringProperty(executablePath);
         this.useAllDatabases = new SimpleBooleanProperty(useAllDatabases);
         this.syncWhenCiting = new SimpleBooleanProperty(syncWhenCiting);
         this.externalStyles = FXCollections.observableArrayList(externalStyles);
         this.currentStyle = new SimpleStringProperty(currentStyle);
+        this.currentStyleType = new SimpleStringProperty(currentStyleType);
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/logic/openoffice/OpenOfficePreferences.java]: Use here the enum directly, e.g. SimpleObjectProperty<StyleType>

File: src/main/java/org/jabref/preferences/JabRefPreferences.java [hunk 2]
```diff
@@ -1281,14 +1283,18 @@ public OpenOfficePreferences getOpenOfficePreferences() {
                 getBoolean(OO_USE_ALL_OPEN_BASES),
                 getBoolean(OO_SYNC_WHEN_CITING),
                 getStringList(OO_EXTERNAL_STYLE_FILES),
-                get(OO_BIBLIOGRAPHY_STYLE_FILE));
+                get(OO_BIBLIOGRAPHY_STYLE_FILE),
+                get(OO_BIBLIOGRAPHY_STYLE_TYPE),
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/preferences/JabRefPreferences.java]: For the enum use StyleType.valueOf(get(OO_BIBLIOGRAPHY_STYLE_TYPE))

as a related example for dealing with enums see GroupsPreferences

File: src/main/java/org/jabref/gui/openoffice/StyleSelectDialogViewModel.java [hunk 3]
```diff
@@ -182,9 +183,40 @@ public void setSelectedTab(Tab tab) {
...
+            }
+        } else {
+            // Jstyle selection
+            openOfficePreferences.setCurrentStyleType("Jstyle");
+            openOfficePreferences.setCurrentStyle(selectedItem.getValue().getStylePath());
         }
     }
+
+    public void initializeSelectedStyle() {
+        if ("CSL".equals(openOfficePreferences.getCurrentStyleType())) {
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/openoffice/StyleSelectDialogViewModel.java]: If you use the enum value in the preferences you can make this a typesafe check

File: src/main/java/org/jabref/gui/openoffice/StyleSelectDialogViewModel.java [hunk 4]
```diff
@@ -182,9 +183,40 @@ public void setSelectedTab(Tab tab) {
     }
 
     public void handleStyleSelection() {
-        CitationStylePreviewLayout selectedLayout = selectedLayoutProperty.get();
-        if (selectedLayout != null) {
-            CSLCitationOOAdapter.setSelectedStyleName(selectedLayout.getDisplayName());
+        if (selectedTab.get().getText().equals("CSL Styles")) {
```
Comment (calixtus) [inline src/main/java/org/jabref/gui/openoffice/StyleSelectDialogViewModel.java]: No magic strings

**Review by Siedlerchr — CHANGES_REQUESTED** [review]
use enum in prefs

**Comment by subhramit** [comment]
Applied review changes

File: src/main/java/org/jabref/logic/openoffice/style/StyleLoader.java [hunk 5]
```diff
@@ -129,17 +136,78 @@ public boolean removeStyle(OOBibStyle style) {
...
+        OOBibStyle defaultStyle = internalStyles.getFirst();
+        openOfficePreferences.setCurrentJStyle(defaultStyle.getPath());
+        openOfficePreferences.setCurrentStyleType(StyleSelectDialogViewModel.StyleType.JSTYLE);
+        return new StyleIdentifier(StyleSelectDialogViewModel.StyleType.JSTYLE, defaultStyle);
+    }
+
+    /**
+     * @param style This can be either a String (for CSL) or OOBibStyle (for JStyle)
+     */
+    public class StyleIdentifier {
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/logic/openoffice/style/StyleLoader.java:173]: You now run into the architecture issues I already talked about in the beginning.

Make this an interface.
Let both CSLStyle and OOStyle implement the interface
In the StyleSelectionViewMOdel you then store he concrete style object 
https://docs.oracle.com/en/java/javase/17/language/pattern-matching-instanceof-operator.html 

As an example you can take a look at the PreviewLayout interface and it's subclasses, e.g. CitationStylePreviewLayout   
The latter one also has a path method 


I think this makes it a bit more cleaner and you don't have to do a cast to object

**Pull request closed without merging**